NAS2D/Renderer/Rectangle.h

```diff
@@ -137,8 +135,8 @@ namespace NAS2D
 			return startPoint() <= point && point < endPoint();
 		}
 
-		// Start point inclusive (x, y), endpoint exclusive (x + width, y + height)
-		// Area in interval notation: [x .. x + width), [y .. y + height)
+		// Start point exclusive (x, y), endpoint exclusive (x + width, y + height)
+		// Area in interval notation: (x .. x + width), (y .. y + height)
```

Comment:
This change is wrong. The original documentation was using [interval notation](https://brilliant.org/wiki/interval-notation/), where square and round brackets have different meaning. Square brackets are inclusive, while round brackets are exclusive.

The choice of inclusive/exclusive endpoints was to make it easy to lay out covering rectangles with no overlap and no space in between. This typically matches how rectangles are used. Consider a rectangle representing screen coordinates for a 640x480 display. The point `{0, 0}` could be the top left pixel, while `{640, 480}` is off the screen.

Reply:
I changed the comments to match what the code was doing. `overlaps` was written to be exclusive: `startPoint() < endPoint()`. vs `contains` which was written as inclusive: `startPoint <= endPoint()`.

Reply:
Check the unit tests. The examples may help.

Both functions use the same criteria. The difference is that `contains` works with `Point` objects, which have no area, whereas `overlaps` works with `Rectangle` objects, which do have area. Comparisons with `endPoint()` use `<` for both of them.

Reply:
And for `overlaps`, pay attention to how the two different `Rectangle` objects are used. The `startPoint()` of each is compared against the `endPoint()` of the other.